src/core/mac/sub_mac.cpp

```diff
@@ -394,13 +394,13 @@ void SubMac::StartCsmaBackoff(void)
 
         if (ShouldHandleTransmitTargetTime())
         {
-            if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) <
-                Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) + mTransmitFrame.mInfo.mTxInfo.mTxDelay -
-                    kCcaSampleInterval - kCslTransmitTimeAhead - kRadioHeaderShrDuration)
+            uint32_t aheadTime      = kCcaSampleInterval + kCslTransmitTimeAhead + kRadioHeaderShrDuration;
+            Time     timerStartTime = Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) - aheadTime;
+            Time     txStartTime    = timerStartTime + mTransmitFrame.mInfo.mTxInfo.mTxDelay;
+
+            if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) < txStartTime)
             {
-                mTimer.StartAt(Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) - kCcaSampleInterval -
-                                   kCslTransmitTimeAhead - kRadioHeaderShrDuration,
-                               mTransmitFrame.mInfo.mTxInfo.mTxDelay);
+                TimerStartAt(timerStartTime, mTransmitFrame.mInfo.mTxInfo.mTxDelay);
```

Comment:
Just noticed something here.

I'm assuming `otPlatRadioGetNow()` and `Timer::GetNow()` might return different values and aren't necessarily always synchronized?

We compare the "timer start time" with `otPlatRadioGetNow()` here, but then seem to use the same time value with `mTimer` (which wouldn't use `otPlatRadioGetNow()`).

Is this correct? Could this potentially lead to an issue? Should we adjust the calculation to convert the start time to `Timer::GetNow()` time?

Thoughts?

Reply:
If this is an issue, we probably should fix it in its own separate PR?

Reply:
The value of `Timer::GetNow()` comes from `otPlatAlarmMicroGetNow()`. In theory, the `otPlatAlarmMicroGetNow()` and `otPlatRadioGetNow()` can use different timer sources. There should be an issue in this case.

I think we can solve this issue here by using relative time. I will push another PR to fix this issue after this PR is merged.

Reply:
Thanks. Sounds good. (I can also fix this, let me know if you want me to submit a fix).

Reply:
I have pushed a PR https://github.com/openthread/openthread/pull/10525 to fix this issue.